src/NuGet.Services.EndToEnd/Support/Clients/GalleryClient.cs

```diff
@@ -156,6 +156,30 @@ public async Task UnlistAsync(string id, string version, ITestOutputHelper logge
             await SendToPackageAsync(HttpMethod.Delete, id, version, logger);
         }
 
+        public async Task SearchPackageODataV2FromDBAsync(string id, bool semver2, ITestOutputHelper logger)
+        {
+            var galleryEndpoint = await GetGalleryUrlAsync(logger);
+            var url = semver2 
+                ? $"{galleryEndpoint}/api/v2/Packages()?$filter=tolower(Id) eq '{id.ToLower()}'&$orderby=Id&semVerLevel=2.0.0"
+                : $"{galleryEndpoint}/api/v2/Packages()?$filter=tolower(Id) eq '{id.ToLower()}'&$orderby=Id";
+            using (var httpClient = new HttpClient())
+            using (var request = new HttpRequestMessage(HttpMethod.Get, url))
+            {
+                request.Headers.Add(ApiKeyHeader, _testSettings.ApiKey);
+
+                using (var response = await httpClient.SendAsync(request))
+                {
+                    await response.EnsureSuccessStatusCodeOrLogAsync(url, logger);
+                    var responseContent = response.Content != null ? await response.Content.ReadAsStringAsync() : string.Empty; 
+                    logger.WriteLine($"The response to the request {url} is {responseContent}");
+                    if(!responseContent.Contains(id))
+                    {
+                        throw new Exception("The expected package id was not found in db.");
```

Comment:
Is that better if the exception message contains the package id?

Reply:
Please throw something more specific that `Exception` like `InvalidOperationException` or just `Assert.Contains`

Reply:
Updated to Assert.Contains(id, responseContent)